makefiles/Makefile.common

```diff
@@ -30,13 +30,55 @@ OPS_LIB_OMPOFFLOAD=-lops_ompoffload
 OPS_LINK=
 CUDART=-lcudart
 
-COMPILER_AVAILABLE := clang gnu intel icx intel_win mic pgi cray xl hip xlp8 aocc
```

Comment:
I think this was a better way of doing things. Why go back to the old if-else structure?

Reply:
You asked me earlier to revert this

Reply:
Perhaps I misunderstood which way - but the point is we should have the generic structure which will work with any makefile extension that is present.